src/System.CommandLine/Parsing/Parser.cs

```diff
@@ -2,56 +2,64 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Collections.Generic;
+using System.Linq;
 
 namespace System.CommandLine.Parsing
 {
     /// <summary>
     /// Parses command line input.
     /// </summary>
-    public class Parser
+    public static class Parser
     {
-        /// <param name="configuration">The configuration on which the parser's grammar and behaviors are based.</param>
-        /// <exception cref="ArgumentNullException"></exception>
-        public Parser(CommandLineConfiguration configuration)
-        {
-            Configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
-        }
-
-        /// <param name="command">The root command for the parser.</param>
-        public Parser(Command command) : this(new CommandLineConfiguration(command))
-        {
-        }
-
         /// <summary>
-        /// Gets the configuration on which the parser's grammar and behaviors are based.
+        /// Parses a list of arguments.
         /// </summary>
-        public CommandLineConfiguration Configuration { get; }
+        /// <param name="command">The command to use to parse the command line input.</param>
+        /// <param name="arguments">The string array typically passed to a program's <c>Main</c> method.</param>
+        /// <param name="configuration">The configuration on which the parser's grammar and behaviors are based.</param>
+        /// <returns>A <see cref="ParseResult"/> providing details about the parse operation.</returns>
+        public static ParseResult Parse(Command command, IReadOnlyList<string>? arguments, CommandLineConfiguration? configuration = null)
```

Comment:
TBH I don't like the fact that we accept `null` arguments. I would be happy to remove it if others think similar of it.

Reply:
No objection from me.

Reply:
The term "arguments" here is potentially confusing because of the `Argument` class. The common name for the `string[]` parameter to `Main` is `args` which is possibly clearer about its intent.

Reply:
Disallowing `null` here makes sense.